maintenance/monitor-component-metrics.md

```diff
@@ -3,43 +3,312 @@ title: Monitor Calico component metrics
 
-### Concepts
+## Concepts
 
-#### About Prometheus
+### About Prometheus
 
 The Prometheus monitoring tool scrapes metrics from instrumented jobs and displays time series data in a visualizer (such as Grafana). For {{site.prodname}}, the “jobs” that Prometheus can harvest metrics from are the Felix and Typha components. 
 
-#### About {{site.prodname}} Felix and Typha components
+### About {{site.prodname}} Felix and Typha components
 
 **Felix** is a daemon that runs on every machine that provides endpoints ({{site.prodname}} nodes). Felix is the brains of {{site.prodname}}. Typha is an optional daemon that extends Felix to scale traffic between {{site.prodname}} nodes and the datastore. **Typha** is used to avoid bottlenecks and performance issues in the datastore when you have over 50 {{site.prodname}} nodes. 
 
 You can configure Felix and/or Typha to provide metrics to Prometheus.
 
+## Before you begin...
 
-### How to
+In this tutorial we assume that you have completed all other introductory tutorials and possess a running Kubernetes cluster with {{site.prodname}}, calicoctl and kubectl installed.
 
-#### Enable Prometheus metrics for and Felix and Typha
+## How to
 
-1. Using the Prometheus documentation, configure one or more [Prometheus servers](https://prometheus.io/docs/introduction/overview/).  
-1. To enable [Felix]({{ site.baseurl }}/reference/felix/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. To enable [Typha]({{ site.baseurl }}/reference/typha/configuration) for metrics, set **PrometheusMetricsEnabled = true**.
-1. If required for Felix and/or Typha, change the default TCP port (9091) for your Prometheus metrics server using the parameter, **PrometheusMetricsPort**.
+This tutorial will go through the necessary steps to implement basic monitoring of {{site.prodname}} with Prometheus.
+1. This section will teach you how to configure {{site.prodname}} to export its own metrics.
+2. In this section you will learn to isolate the resources and create necessary permissions for future steps.
+3. In this section you will configure Prometheus and create its instance in the cluster.
+4. You will be able to visit Prometheus dashboard and create a simple graph.
 
-#### Best practices
+
+### 1. Configure Calico to enable metrics reporting
+#### Felix configuration
+Felix prometheus metrics are **disabled** by default. You have to manually change your Felix configuration (**prometheusMetricsEnabled**) via calicoctl in order to use this feature.
+
+> **Note**: A comprehensive list of configuration values can be [found at this link]({{ site.baseurl }}/reference/felix/configuration).
+   {: .alert .alert-info}
+
+```bash
+calicoctl patch felixConfiguration default  --patch '{"spec":{"prometheusMetricsEnabled": true}}'
+```
+You should see an output like below:
+```
+Successfully patched 1 'FelixConfiguration' resource
+```
+
+##### **Creating a service to expose Felix metrics**
```

Comment:
Does it make sense to have a service for Felix metrics?  

From https://docs.projectcalico.org/reference/felix/prometheus we can see that some of the metrics are local to the particular felix you've connected to.  

A service is going to load-balance your connection to a random felix, so if you were viewing this as a data stream you'll see it bounce around between the particular values sent by each different felix.  

Is that desirable?

Reply:
This service creation is intentional, service will be used in order to resolve the endpoints that are connected to it.
Prometheus will not use the service to collect any metrics.
Metrics will be collected from each Calico pod individually.

Reply:
@lwr20 It is possible to get metrics without using services, Using kubernetes_sd_config and pod selection with a fairly complex regex selector should do the trick.
It might make the tutorial a bit harder to understand for people with less experience with regex and Prometheus.
Would you like to see the implementation, it might help with the decision?

Reply:
No need to show, I can believe it :)  Would a note explaining that the service is not used directly, but to create a list of endpoints to get metrics from be better?

Reply:
Something like this?
(add to line 62 and 105)
It is worth mentioning that this service provides endpoints and the port that Typha is using and it will not be used to connect to the instance and collect metrics.